### PR TITLE
Feature/187 automated api tests

### PR DESCRIPTION
**Pull request by mariannemyers**
See #7 

Pulling existing PR 11 over from Bitbucket.  

Will follow up with existing comments copy/pasted over here, sourcing the appropriate author.

### CODE REVIEW

**Comment by mariannemyers** [comment]
@patrickmcelwee: Playing with this. It seems very, very nice. I like all the clean-up that it prompted in grove-node.

Does this eliminate the need for the "small VueJS page" cited in the README? Also, I'd like to add some documentation about this test suite in the README.

**Comment by mariannemyers** [comment]
@patrickmcelwee: The tests themselves run great!

I took a stab at running this against MarkLogic in Docker (on a separate branch) - add-docker-to-187

For the moment, at least, grove-node is running separately, hitting the Docker container on port 8063. It all requires that port 8063 is free on the host. It also does not allow separate Docker containers for individual groups of tests (which would increase test separation).

I still need to host the Docker image on a registry.

And some thought would be helpful on how to test different 1) MarkLogic versions, 2) grove-ml-gradle or other ml-gradle configs, and 3) grove-node versions.

File: spec/ping-api.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
+{
+  "openapi": "3.0.0",
+  "servers": [],
+  "info": {
+    "description": "This defines a simple API for checking if middleTier is alive.",
+    "version": "0.10.0-beta",
```
Comment (mariannemyers) [inline spec/ping-api.json]: @patrickmcelwee: we should change all these version to 1.0.0-rc.2

**Comment by mariannemyers** [comment]
@grtjn: Note to self: need to revisit this PR, and go over all the feedback from Patrick.

File: spec/ping-api.json (hunk 1 above)
Comment (grtjn) [reply spec/ping-api.json]: yes, will do..

**Comment by grtjn** [comment]
I rebased against latest origin/development to get rid of merge conflicts, and updated the versions to rc2. I'll revise the section on testing, so we can finally close this PR..

**Comment by grtjn** [comment]
Ran a test against a grove middle-tier, and it worked like a charm after fixing one last issue with the new search tests. Updated README to include details on testing with Jest.

I think it is ready for merging. @mariannemyers, care to glance over?

**Comment by grtjn** [comment]
i need to push a tweak to the grove-ml-gradle 2-tier branch, there was a flaw in there too..

**Comment by grtjn** [comment]
Ran tests against current grove-node middle-tier, and polished it until it seemed to run fine. Merging this now. A PR for grove-node is imminent..

**Pull request merged**